File: Leetcode/53. Maximum Subarray/solution.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if nums:
            maxi = nums[0]
            cur = nums[0]

            if len(nums) == 1:
                return nums[0]

            for num in nums[1:]:
                if cur < 0:
                    cur = num
                else:
                    cur += num
                maxi = max(cur, maxi)

            return maxi

        return 0

File: Leetcode/53. Maximum Subarray/solution.rs
use std::cmp::max;

impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut maxes = nums[0];
        let mut ans = nums[0];

        for i in 1..nums.len() {
            maxes = max(maxes + nums[i], nums[i]);
            ans = max(maxes, ans);
        }

        ans
    }
}
